src/geneweaver/api/schemas/apimodels.py

```diff
@@ -1,21 +1,35 @@
 """Models for API requests."""
-from typing import Iterable, Optional
+from typing import Iterable, List, Optional
 
 from geneweaver.core.enum import GeneIdentifier, Species
 from pydantic import BaseModel
 
 
-class GeneIdMappingResp(BaseModel):
-    """Model for gene id mapping."""
+class GeneIdHomologResp(BaseModel):
+    """Model for homolog gene id mapping."""
 
     gene_ids_map: list[dict]
 
 
-class GeneIdMappingReq(BaseModel):
-    """Model for gene id mapping request."""
+class GeneIdHomologReq(BaseModel):
+    """Model for homolog gene id mapping request."""
 
     source_ids: Iterable[str]
     target_gene_id_type: GeneIdentifier
     source_gene_id_type: Optional[GeneIdentifier] = None
     target_species: Optional[Species] = None
     source_species: Optional[Species] = None
+
+
+class GeneIdMappingResp(BaseModel):
+    """Model for gene id mapping."""
+
+    gene_ids_map: list[dict]
+
+
+class GeneIdMappingReq(BaseModel):
+    """Model for gene id mapping request."""
+
+    source_ids: List[str]
+    target_gene_id_type: GeneIdentifier
+    target_species: Species
```

Comment:
I think "source_species" or just "species" would be more correct. These in a non-cross species endpoint, so everything will be within a single species.